src/VisualStudio/Core/Def/LanguageService/AbstractCreateServicesOnTextViewConnection.cs

```diff
@@ -69,14 +73,29 @@ void IWpfTextViewConnectionListener.SubjectBuffersDisconnected(IWpfTextView text
         }
 
         private void InitializeServiceOnDocumentOpened(object sender, DocumentEventArgs e)
+            => InitializeServiceForDocument(e.Document);
+
+        private void InitializeServiceOnAlreadyOpenedDocuments(ImmutableArray<DocumentId> documentIds)
+        {
+            foreach (var id in documentIds)
+            {
+                var document = Workspace.CurrentSolution.GetDocument(id);
+                if (document != null)
+                {
+                    InitializeServiceForDocument(document);
+                }
+            }
+        }
+
+        private void InitializeServiceForDocument(Document document)
         {
-            if (e.Document.Project.Language != _languageName)
+            if (document.Project.Language != _languageName)
             {
                 return;
             }
 
             var token = _listener.BeginAsyncOperation(nameof(InitializeServiceForOpenedDocumentOnBackgroundAsync));
-            InitializeServiceForOpenedDocumentOnBackgroundAsync(e.Document).CompletesAsyncOperation(token);
+            InitializeServiceForOpenedDocumentOnBackgroundAsync(document).CompletesAsyncOperation(token);
```

Comment:
Where is the result of this observed? Are we hiding exceptions?

Reply:
you are right. I have changed it to use async task queue which handles exceptions and report NFWs.